### PR TITLE
EXL-300 Fixed self test queries and emaintenance.class.php

### PR DESCRIPTION
**Pull request by zsso**
- Fixed query in cartridgeitems_invalid_customfields.php
- Fixed query in printers_with_old_em_reading_data.php
- Fixed 'effective_date_field', 'total2_black_field', 'total2_color_field' variables in emaintenance.class.php

### CODE REVIEW

File: plugin/inc/dataintegrity_tests/cartridgeitems_invalid_customfields.php [hunk 1]
```diff
@@ -8,9 +8,9 @@
           and (   coalesce(ci.ref, '') = ''
                or coalesce(ci.mercury_code_field, '') = ''
                or coalesce(ci.compatible_mercury_codes_field, '') = ''
-               or coalesce(ci.atc_field, 0) = 0 
-               or coalesce(ci.life_coefficient_field, 0) = 0 
-               or coalesce(ci.supported_types_field, 0) = 0)
+               or coalesce(ci.atc_field, '0') = '0' 
+               or coalesce(ci.life_coefficient_field, '0') = '0' 
+               or coalesce(ci.supported_types_field, '0') = '0')
```
Comment (hupupu) [inline plugin/inc/dataintegrity_tests/cartridgeitems_invalid_customfields.php:13]: The atc and life_coefficient fields should be decimals or ints, the supported_types is varchar, so the check should also chanhe
```suggestion
               or coalesce(ci.atc_field, 0) = 0 
               or coalesce(ci.life_coefficient_field, 0) = 0
               or coalesce(ci.supported_types_field, '') = '')
```

Also: I also observed somewhere that you made a similar change (a made a comment), I don't know what happened with that.
The best would probably be to re-check the type of every customfield (to match with the type in v2)

File: plugin/inc/dataintegrity_tests/printers_with_old_em_reading_data.php [hunk 2]
```diff
@@ -61,7 +61,19 @@ function iservice_custom_command_check_em_csv()
...
+				    if(
+				        datediff(curdate(), p.last_read_field) > 15000,
+				        'infinite',
+				        if(
+				            datediff(curdate(), p.last_read_field) > 14000,
+				            'unknown',
+				            lpad(datediff(curdate(), p.last_read_field), 5, '0')
+				        )
+				    )
+            ) as days_since_last_read
```
Comment (hupupu) [inline plugin/inc/dataintegrity_tests/printers_with_old_em_reading_data.php:76]: we have 2 'unknown' cases. change the first to 'empty'

**Review by hupupu — CHANGES_REQUESTED** [review]

File: plugin/inc/dataintegrity_tests/cartridgeitems_invalid_customfields.php (hunk 1 above)
Comment (zsso) [reply plugin/inc/dataintegrity_tests/cartridgeitems_invalid_customfields.php]: Fixed.

File: plugin/inc/dataintegrity_tests/printers_with_old_em_reading_data.php (hunk 2 above)
Comment (zsso) [reply plugin/inc/dataintegrity_tests/printers_with_old_em_reading_data.php]: Done.

**Review by hupupu — APPROVED** [review]

**Pull request merged**